mne/preprocessing/maxwell.py

```diff
@@ -339,47 +371,72 @@ def _sss_basis(origin, coils, int_order, ext_order):
+            grads_in_out = list()
+            # Same spherical harmonic is used for both internal and external
+            sph = _sph_harm(order, degree, c_sp[1], c_sp[2], norm=False)
+            norm = _sph_harm_norm(order, degree)
+            sph *= norm
+            if degree <= int_order:
+                S_in_out.append(S_in)
+                # Compute complex gradient for all integration points
+                grads_in_out.append(_grad_in_comps(order, degree, sph, norm,
+                                                   c_sp[0], c_sp[1], c_sp[2]))
+            if degree <= ext_order:
+                S_in_out.append(S_out)
+                grads_in_out.append(_grad_out_comps(order, degree, sph, norm,
+                                                    c_sp[0], c_sp[1], c_sp[2]))
+            for spc, grads in zip(S_in_out, grads_in_out):
+                # We could convert to real at the end, but it's more efficient
+                # to do it now
+                grads_pos_neg = [_sh_complex_to_real(grads, order)]
+                orders_pos_neg = [order]
+                # Deal with the negative orders
+                if order > 0:
+                    # it's faster to use the conjugation property for
+                    # our normalized spherical harmonics than recalculate...
+                    grads_pos_neg.append(_sh_complex_to_real(
+                        _sh_negate(grads, order), -order))
+                    orders_pos_neg.append(-order)
+                for gr, oo in zip(grads_pos_neg, orders_pos_neg):
+                    # Gradients dotted w/integration point normals and weighted
+                    gr = wcoils * np.einsum('ij,ij->i', gr, ncoils, order='C')
+                    # For order/degree, sum over each sensor's integration pts
+                    # for pt_i in range(0, len(int_lens) - 1):
+                    #  int_pts_sum = \
+                    #    np.sum(all_grads[int_lens[pt_i]:int_lens[pt_i + 1]])
+                    #  spc[pt_i, deg ** 2 + deg + oo - 1] = int_pts_sum
+                    vals = np.bincount(bins, weights=gr, minlength=len(counts))
+                    spc[:, _deg_order_idx(degree, oo)] = -vals
+
+    # Scale magnetometers
+    S_tot *= coil_scale[:, np.newaxis]
     return S_tot
 
 
-def _alegendre_deriv(degree, order, val):
+def _deg_order_idx(deg, order):
+    """Helper to get the index into S_in or S_out given a degree and order"""
+    return deg ** 2 + deg + order - 1
+
+
+def _alegendre_deriv(order, degree, val):
     """Compute the derivative of the associated Legendre polynomial at a value.
 
     Parameters
     ----------
-    degree : int
-        Degree of spherical harmonic. (Usually) corresponds to 'l'
     order : int
-        Order of spherical harmonic. (Usually) corresponds to 'm'
+        Order of spherical harmonic. (Usually) corresponds to 'm'.
+    degree : int
+        Degree of spherical harmonic. (Usually) corresponds to 'l'.
```

Comment:
Reason for swapping these throughout? Was it because I had them that way in the _sph_harm function? I think I messed up the order there... The best way is probably to have degree and then order cause it's usually read as "l, m" when both have to be specified